### PR TITLE
Fixed BreakingESP and LoginMessage

### LINKED ISSUE #1214: LoginMessage should use onUpdate() instead of a custom movement event
Makes 0 sense to post a custom movement update when you're going to return if mc.player is null anyways

### LINKED ISSUE #1222: BreakingESP warning setting doesn't do anything
Tested with self / obby / pick all toggled on / off

### PR DESCRIPTION
**Pull request by wnuke**
**Describe the pull**
This is a pull to fix #1222 and fix #1214 

**Describe how this pull is helpful**
This pull changes BreakingESP too fix settings, re-add warning text to center of screen and make the module more user configurable.

**Additional context**
None.

### CODE REVIEW

File: src/main/java/me/zeroeightsix/kami/module/modules/chat/LoginMessage.kt [hunk 1]
```diff
@@ -40,16 +40,14 @@ class LoginMessage : Module() {
             return
         }
         MessageSendHelper.sendChatMessage("$chatName Found '&7loginmsg.txt&f'!")
-
     }
 
-    @EventHandler
-    private val localPlayerUpdateEvent = Listener(EventHook { event: LocalPlayerUpdateEvent? ->
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/chat/LoginMessage.kt:47]: Can you delete LocalPlayerUpdateEvent and where it posts the event in ForgeEventProcessor please?

File: src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt [hunk 2]
```diff
@@ -25,9 +27,14 @@ import net.minecraft.util.math.BlockPos
         category = Module.Category.RENDER
 )
 class BreakingESP : Module() {
-    private val ignoreSelf = register(Settings.b("IgnoreSelf", false))
+    private val espSelf = register(Settings.b("ESPSelf", true))
+    private val warnSelf = register(Settings.b("WarnSelf", false))
     private val obsidianOnly = register(Settings.b("ObsidianOnly", false))
     private val warning = register(Settings.b("Warning", false))
+    private val warningProgress = register(Settings.integerBuilder("Warning Progress").withMinimum(0).withValue(4).withMaximum(9).build())
+    private val chatWarn = register(Settings.b("Chat Warning", true))
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt:35]: should be off by default imo

File: src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt [hunk 3]
```diff
@@ -25,9 +27,14 @@ import net.minecraft.util.math.BlockPos
...
+    private val screenWarn = register(Settings.b("Title Warning", true))
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt:36]: This should be "HudText" or something

File: src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt [hunk 4]
```diff
@@ -70,19 +81,37 @@ class BreakingESP : Module() {
...
     }
 
+    private var warn = false
+    private var delay = 0
+
+    override fun onRender() {
+        if (screenWarn.value && warn) {
+            if (delay++ > 100) warn = false
+            val textRenderer = Wrapper.fontRenderer
+            val divider = DisplayGuiScreen.getScale()
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt:91]: auto gui scale returns 0 so you should have a safety for that iirc

File: src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt [hunk 5]
```diff
@@ -70,19 +81,37 @@ class BreakingESP : Module() {
...
     private val blockBreaklistener = Listener(EventHook { event: BlockBreakEvent ->
         if (mc.player == null || mc.player.getDistanceSq(event.position) > range.value * range.value) return@EventHook
         val breaker = mc.world.getEntityByID(event.breakId) ?: return@EventHook
-        if (ignoreSelf.value && breaker == mc.player) return@EventHook
         if (event.progress in 0..9) {
-            breakingBlockList.putIfAbsent(event.breakId, Triple(event.position, event.progress, false))
+            var render = false
+            if (mc.player != breaker || espSelf.value) {
+                render = true
+            }
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt:104]: you can do 
```kotlin
val render = if (mc.player != breaker || epsSelf.value) {
    true
} else {
    false
}
```

File: src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt (hunk 2 above)
Comment (wnuke) [reply src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt:35]: makes sense

File: src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt (hunk 4 above)
Comment (wnuke) [reply src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt:91]: Oh, good to know, ill change that.

File: src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt (hunk 5 above)
Comment (wnuke) [reply src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt:104]: forgot about that, will change

**Comment by dominikaaaa** [comment]
btw we don't use spaces in setting names, please change that

File: src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt (hunk 3 above)
Comment (wnuke) [reply src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt]: changed to HUD Warning

File: src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt (hunk 2 above)
Comment (wnuke) [reply src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt]: Which warning type should be on by default? Currently chat warning is the only one that defaults to on

Comment (dominikaaaa) [reply src/main/java/me/zeroeightsix/kami/module/modules/render/BreakingESP.kt]: The visual warning

**Review by dominikaaaa — APPROVED** [review]

**Pull request closed without merging**